### PR TITLE
Syft config

### PR DESCRIPTION
**Pull request by vkkhare**
Pr adds #53 #37

### CODE REVIEW

File: syftlib/src/main/java/org/openmined/syft/device/repositories/NetworkStatusRepository.kt [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import java.util.Random
+
+private const val TAG = "NetworkStateRepository"
+
+private const val SPEED_BUFFER_WINDOW = 20
+private const val SPEED_MULTIPLICATION_FACTOR = 10
+private const val MAX_SPEED_TESTING_BYTES = MB * 8
+
+@ExperimentalUnsignedTypes
+class NetworkStatusRepository(private val configuration: SyftConfiguration)  {
```
Comment (mccorby) [inline syftlib/src/main/java/org/openmined/syft/device/repositories/NetworkStatusRepository.kt:28]: I would have put `NetworkStatusRepository` and `SleepStatusRepository` in a separate Android module and leave the library to define the contract.
But I understand it's easier like this. We can refactor later if we think we need to.

Comment (vkkhare) [reply syftlib/src/main/java/org/openmined/syft/device/repositories/NetworkStatusRepository.kt:28]: I was thinking about separating them but the syft worker would still need to call out these test before running the cycle. The syft worker will be needing device monitor so I am not sure if syft.kt could remain independent of android code

File: syftlib/src/main/java/org/openmined/syft/monitor/network/NetworkStatusRepository.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import io.reactivex.Completable
+import io.reactivex.Single
+import org.openmined.syft.domain.SyftConfiguration
+
+private const val TAG = "NetworkStateRepository"
+
+
+@ExperimentalUnsignedTypes
+class NetworkStatusRepository(private val configuration: SyftConfiguration)  {
+    private val cacheService = NetworkStatusCache(configuration.cacheTimeOut)
```
Comment (jacorbachosky) [inline syftlib/src/main/java/org/openmined/syft/monitor/network/NetworkStatusRepository.kt:12]: These could be injected in the constructor.

File: syftlib/src/main/java/org/openmined/syft/monitor/DeviceMonitor.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.monitor.battery.BatteryStatusRepository
+import org.openmined.syft.monitor.network.NetworkStatusRepository
+
+@ExperimentalUnsignedTypes
+class DeviceMonitor(
+    syftConfig: SyftConfiguration
+) {
+
+    private val networkStatusRepository =
```
Comment (jacorbachosky) [inline syftlib/src/main/java/org/openmined/syft/monitor/DeviceMonitor.kt:13]: These could be injected in the constructor.

Comment (vkkhare) [reply syftlib/src/main/java/org/openmined/syft/monitor/DeviceMonitor.kt:13]: Yeah will open up a static function to create the instance using injection

**Pull request merged**